src/main/resources/resource/Solr/core1/conf/managed-schema

```diff
@@ -414,6 +414,10 @@
       <tokenizer class="solr.WhitespaceTokenizerFactory"/>
     </analyzer>
   </fieldType>
+  
+  <!-- Adding a dense vector field for knn search to do cosine similarity ( 384 dimensions and cosine similarity matches with al-->
+  <fieldType name="knn_vector" class="solr.DenseVectorField" vectorDimension="384" similarityFunction="cosine" knnAlgorithm="hnsw" hnswMaxConnections="10" hnswBeamWidth="40"/>
```

Comment:
This is what I ended up doing on my branch as well, but I was wondering if there were a better way to accomplish it. Tying ourselves to a fixed vector dimension limits the models we can support with Solr, is there a way to dynamically define the type according to the vector dimension? If not, can we define multiple types for each of the most common dimensions? Also, do we want to use cosine similarity or can we force normalization before indexing so we can use the faster dot product on search?

Reply:
Good news is that we can programmatically add fields to the schema.  I chose 384 dimensions and cosine similarity because that lines up with the models that are commonly used. ( miniLM-L6-V2 ). https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2. 
Cosine similarity is much more popular compared to dot product similarity.